### PR TITLE
Added a scoring-test

### PR DESCRIPTION
**Pull request by oyugirachel**
Things to test in the scoring function are:

1.Checking the two present cards
2.If the player snaps
3.If the player does not snap
In both cases the score should either decrease or increase

Currently  the last two cases pass , more iteration is being done to make use of the first case

### CODE REVIEW

**Comment by oyugirachel** [comment]
Due to the shuffling of the deck, and the data may change consequently when the program is run, I can use golang template in the golden file and pass the data to AssertWithTemplate.

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		{
+			name:         "sameCards",
+			presentCards:"King of Spades, King of Diamonds",
+			args:         args{snap: false},
+		},
+		
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scoring(tt.args.snap)
```
Comment (dtrudg) [inline main_test.go]: The test needs to check that `scoring(...)` has the correct effect. If it's modifying package level variables you need to check they are changed in the correct way.

You also need to setup the current 2 cards before calling `scoring`.

**Review by dtrudg — CHANGES_REQUESTED** [review]
A nice start thinking about what combinations of things need to be tested here.

We need to look at how we set up preconditions for the test, and check the result. Let's talk about it in today's call.

File: main.go [hunk 2]
```diff
@@ -129,25 +129,25 @@ func drawCard(done chan bool, cards []deck.Card, ticker *time.Ticker) {
...
 // scoring function
-func scoring(snap bool) {
+func scoring(snap bool) int{
 
 	if snap {
 		if presentCards[0].Rank == presentCards[1].Rank {
 			score++
 			fmt.Println("\nYour score is:", score)
-			return
+			return score
```
Comment (dtrudg) [inline main.go]: It's a little bit confusing to return the global variable `score` here. Usually if we return something we want to try and return something that is as limited to the scope of the function as possible.

For example, we might think about `scoring` returning the change in score that needs to be made (i,e +1/0/-1) and applying that to the global game score where we call `scoring` instead.

File: main_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"github.com/oyugirachel/deck"
+	"testing"
+	
+)
+
+var (
+	ace   = deck.Card{Suit: deck.Spade, Rank: deck.Ace}
```
Comment (dtrudg) [inline main_test.go]: All of the cards you setup here are in the same Suit. This means that your tests wouldn't catch issue #1 - where the game rules say we match on rank, so cards of the same rank but different suits still count for a snap.

Think about adding a sub-test or tests to cover issue 1

File: main_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			score: -1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cards := deck.New(deck.Deck(1), deck.Shuffle)
+			lastCard = 1
+			presentCards[0] = presentCards[1]
+			presentCards[1] = cards[lastCard]
+			presentCards = tt.cards
```
Comment (dtrudg) [inline main_test.go:76]: This line sets the value of presentCards, so you shouldn't need to use the lines above.

Check carefully what the scoring function uses. Does it depend on `lastCard` or `cards` at all? Do you need to set those up in this test?

**Review by dtrudg — CHANGES_REQUESTED** [review]
The tests don't quite work. This is likely because, as we discussed, the `scoring` function may change the global `score` variable each time it is called. `score` can then have a different starting value between each test that is run, as the previous test may have changed it.

We usually want to make tests as independent as possible, so it isn't important if they are run in a particular order. We can't always do it completely, but in this situation you can improve things.

To make the tests independent you could either:

1. Fix `score` to a specific value at the start of each test.
2. Change the `scoring` function so it returns the _change in score_ (+1/0/-1) only, and does not alter the global game score.

**Review by dtrudg — APPROVED** [review]

**Pull request merged**